src/services/user.service.js

```diff
@@ -0,0 +1,45 @@
+import { User } from '../models/user.js';
+import { v4 as uuidv4 } from 'uuid';
+import { emailService } from './email.services.js';
+import { ApiError } from '../exceptions/api.error.js';
+
+const getByEmail = (email) => {
+  return User.findOne({ where: { email } });
+};
+
+const registerUser = async (name, email, password) => {
+  const activationToken = uuidv4();
+  const isUserExist = await getByEmail(email);
+
+  if (isUserExist) {
+    throw ApiError.notFound();
```

Comment:
This line throws an ApiError.notFound() when a user already exists, which might not be the correct error type. Consider using an error type that indicates a conflict or that the user already exists.